metpy/calc/tools.py

```diff
@@ -330,3 +332,142 @@ def log_interp(x, xp, fp, **kwargs):
+            if heights is not None:
+                bottom_height = log_interp(bottom_pressure, p, heights)
+            else:
+                bottom_height = pressure_to_height_std(bottom_pressure)
+        # in height units
+        if bottom.dimensionality == {'[length]': 1.0}:
+            bottom_height = bottom
+            if heights is not None:
+                bottom_pressure = np.interp(np.array([bottom_height.m]), heights,
+                                            p)[0] * p.units
+            else:
+                bottom_pressure = height_to_pressure_std(bottom_height)
+
+    # Top of layer calculations
+
+    # Depth is in pressure
+    if depth.dimensionality == {'[length]': -1.0, '[mass]': 1.0, '[time]': -2.0}:
+        top_pressure = bottom_pressure - depth
+        if heights is not None:
+            top_height = log_interp(top_pressure, p, heights)
+        else:
+            top_height = pressure_to_height_std(top_pressure)
+
+    # Depth is in height
+    if depth.dimensionality == {'[length]': 1.0}:
+        top_height = bottom_height + depth
+        if heights is not None:
+            top_pressure = np.interp(np.array([top_height.m]), heights, p)[0] * p.units
+        else:
+            top_pressure = height_to_pressure_std(top_height)
+
+    # Handle top or bottom values that are invalid
+    if bottom_pressure > p[0]:
+        raise ValueError(
+            'Bottom of layer pressure is greater than the largest pressure in data.')
+    if top_pressure < p[-1]:
+        raise ValueError('Top of layer pressure is less than the lowest pressure in data.')
+    if bottom_pressure < top_pressure:
+        raise ValueError(
+            'Pressure at the top of the layer is greater than that at the bottom.')
+
+    ret = []  # returned data variables in layer
+
+    # Ensure pressures are sorted in ascending order
+    sort_inds = np.argsort(p)
+    p = p[sort_inds]
+
+    # Mask based on top and bottom pressure
+    inds = (p <= bottom_pressure) & (p >= top_pressure)
+    p_interp = p[inds]
+
+    # Interpolate pressures at bounds if necessary and sort
+    if interpolate:
+        # If we don't have the bottom or top requested, append them
+        if top_pressure not in p_interp:
+            p_interp = np.sort(np.append(p_interp, top_pressure)) * p.units
+        if bottom_pressure not in p_interp:
+            p_interp = np.sort(np.append(p_interp, bottom_pressure)) * p.units
+
+    for datavar in args:
+        # Ensure that things are sorted in ascending order
+        datavar = datavar[sort_inds]
+
+        if interpolate:
+            # Interpolate for the possibly missing bottom/top values
+            datavar_interp = log_interp(p_interp, p, datavar)
+            datavar = datavar_interp
+        else:
+            datavar = datavar[inds]
+
+        ret.append(datavar[::-1])
+
+    # If heights are given, return the height range as well as pressure range
+    if heights is not None:
```

Comment:
I argued for it until we got to:
```python
sbcape(*get_layer(p, t, td, heights=h))
```
I'd really like to preserved that, and including heights would break that.